cpp/src/arrow/compute/exec/test_nodes.h

```diff
@@ -0,0 +1,52 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <string>
+
+#include "arrow/compute/exec/options.h"
+#include "arrow/compute/exec/test_util.h"
+
+namespace arrow {
+namespace compute {
+
+struct ConcatNodeOptions : public ExecNodeOptions {
```

Comment:
Perhaps this should be named `BackpressureOptions` and moved to `arrow/compute/exec/options.h`, while `ConcatNodeOptions` should derive from it or include it as a member. Not critical for this PR, but seems useful.

Reply:
We can do this as follow up if needed - Let's try to keep the PR as is (unless there is something that we really don't like and want to change in the original PR).

Reply:
Hmm, the concat node is only meant as an example for exec node developers right now.  There's no way to actually use it in a deployed environment.  So let's wait until there is a real node that needs backpressure options and then we can move it.

Reply:
And, actually, there is already `arrow::compute::BackpressureOptions` in `exec/options.h` for the sink node.  I think I'll just rip out the concat node for now since it isn't used in tests anymore.